### ISSUE TITLE
Add a new top-level section about error handling

### ISSUE THREAD

**Issue by Witiko** [Witiko/fast-regression-testing#13]
0Currently, we discuss error handling in Section 4 about Implementation. However, we only discuss batch splitting and not the options `--fail-fast` and `--update-tests`, which help illustrate the importance of using an expressive language such as Python for the implementation and which help draw parallels between our framework and l3build. Therefore, we should:
- Create a new top-level section about error handling set between sections 3 (Techniques) and 4 (Implementation).
- Move the subsection about batch splitting from Section 4 (now 5) to the new section.
- Add subsections about the options `--fail-fast` and `--update-tests` to the new section.
  For `--fail-fast`, discuss the preferential treatment of added and modified test files from https://github.com/Witiko/markdown/commit/feafe9b9c02bb391c8f863597d222448b5a940fd.
- In Section 4 (now 5), mention the options `--fail-fast` and `--update-tests` as reasons for implementing our framework in Python.
- Update Section 4 (now 5) and Listing 1 to show that `test.sh` supported updating test files from the onset, whereas the support for failing fast and batch splitting were added in Markdown 3.0.0.
- In Section 7.1 (now 8.1, l3build), mention `--update-tests` as a common feature of our framework and l3build.
- In Conclusion, mention error handling as one of the lessons learned.

After the addition of the new section, the introduction of the multiprocessing and batching techniques in Section 3 and the description of batch size limiting in Section 4 (now 5) are now quite distant, although they are highly related. Therefore, we should also:
- Move subsection about batch size limiting from Section 4 (now 5) to the end of Section 3.

This makes Section 4 (now 5) consist of only one subsection (Before and after Markdown 3.0.0), so we should also:
- Remove first paragraph of Section 4 (now 5) and remove the heading ``Before and after Markdown 3.0.0''.
